### PR TITLE
Add some docs in Nova Editor types

### PR DESCRIPTION
**Pull request by chriskrycho**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://docs.nova.app/api-reference/language-client/>

---

In addition to the primary change of adding docs, I fixed on type issue I spotted in the tests and also tackled a couple lint rules that were being flagged up.

Also, for the maintainer: It *looks* to me (but I have not double checked, so I could be wrong?) like Nova’s types here should actually just use the types for `LanguageClient` etc. straight from the LSP package?

### CODE REVIEW

**Comment by chriskrycho** [comment]
Whoa! I'm somewhat surprised this is the first time `vscode-languageserver-types` is a dependency!

File: types/nova-editor-node/index.d.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@
 /// <reference no-default-lib="true"/>
 /// <reference lib="es2020" />
 /// <reference lib="WebWorker" />
+/// <reference types="vscode-languageserver-types" />
```
Comment (jakebailey) [inline types/nova-editor-node/index.d.ts:8]: How is this working? Doesn't this package export its types, not globally declare them?

Comment (chriskrycho) [reply types/nova-editor-node/index.d.ts:8]: Oh, good catch; I just forgot to remove this. It does indeed, and the relevant line is the one further down the file that does it with an `= import('vscode-languageserver-types').TheType` instead.

```suggestion
```

**Comment by chriskrycho** [comment]
Haven't forgotten about this, but it has been lower priority; hopefully will get it fixed up either later today or early next week.

**Pull request closed without merging**